web/controllers/api/key_controller.ex

```diff
@@ -23,7 +23,18 @@ defmodule HexWeb.API.KeyController do
   end
 
   def create(conn, params) do
-    case Key.create(conn.assigns.user, conn.params) |> HexWeb.Repo.insert do
+    result =
+      HexWeb.Repo.transaction(fn ->
+        case Key.create(conn.assigns.user, conn.params) |> HexWeb.Repo.insert do
+          {:ok, key} ->
+            HexWeb.AuditLog.create(conn.assigns.user, "key.generate", key) |> HexWeb.Repo.insert!
```

Comment:
:+1:

I'm considering doing one more addition which would clean up the controllers a little bit, basically:

```elixir
# controller_helpers.ex

# decent docs explaining this should be running in a transaction
def audit(conn, action, opts) do
 ...
end

# key_controller.ex
audit("key.generate", key)
```

what do you think?